pkg/api/rest/delete.go

```diff
@@ -69,14 +69,13 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 		// only a shorter grace period may be provided by a user
 		if options.GracePeriodSeconds != nil {
 			period := int64(*options.GracePeriodSeconds)
-			if period > *objectMeta.DeletionGracePeriodSeconds {
-				return false, true, nil
+			newDeletionTimestamp := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
```

Comment:
It should always be possible to shorten the grace period seconds, regardless of where timestamp is.   I.e., a user who does:

DELETE 30s at t1
DELETE 15s at t2

should end up with a resource with grace period 15s with deletion timestamp t1.